src/utils/event_gauge.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <atomic>
+#include <cstdlib>
+#include <memory>
+
+namespace Statistics {
+using Event = uint64_t;
+using Value = uint64_t;
+using Gauge = std::atomic<Value>;
+
+class EventGauges {
+ public:
+  explicit EventGauges(Gauge *allocated_gauges) noexcept : gauges_(allocated_gauges) {}
+
+  auto &operator[](const Event event) { return gauges_[event]; }
+
+  const auto &operator[](const Event event) const { return gauges_[event]; }
```

Comment:
Is it necessary to have both with and without const? I am really not sure.

Reply:
I think it is good as then when you have const variable, you can't call function to get counter

Reply:
But you can const_cast it or something?